From edcafad9548ce5ecb1023bdc38be71c41326eacd Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 17 Feb 2021 00:49:22 +0200
Subject: [PATCH] [SW-29740] gaudi: security: add mme1_qm protection

The MME1_QM's interface clock (LBW) is disabled by default and thus to
access it, it must be enabled. Another option is to protect MME1_QM from
access and thus prevent any "non-clocked" access.
Use additional RR to cover the MME1_QM.

Change-Id: I422abfb1e67377ad0a7220bf4a9a7b041e6cc394
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 9bd5bcf..62385b7 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1167,6 +1167,7 @@
 }
 
 enum gaudi_sec_lbw_rrs {
+	LBW_RR_MME1_QM,
 	LBW_RR_MME3_QM_SRAM_CFG,
 	LBW_RR_CPU_TSTAMP,
 	LBW_RR_HBM_GICBGN,
@@ -1194,6 +1195,11 @@
 	 * meaning - not inclusive, so we need to provide it the min address
 	 * below the forbidden one and max address above the forbidden one.
 	 */
+	/* MME1_QM: RR(LBW_RR_MME1_QM) */
+	min = mmMME1_QM_BASE - 1;
+	max = mmMME2_ACC_BASE;
+	preboot_secure_lbw_rr(LBW_RR_MME1_QM, min, max);
+
 	/* MME3_QM, SRAM_CFG up to SIF: RR(LBW_RR_MME3_QM_SRAM_CFG) */
 	min = mmMME3_QM_BASE - 1;
 	max = mmSIF_RTR_0_BASE;
